### PR TITLE
Update tags

### PR DESCRIPTION
**Pull request by markdboyd**
Related to https://github.com/cloud-gov/s3-broker/issues/57

## Changes proposed in this pull request:

- Update tests to use `github.com/onsi/ginkgo/v2`
- Update logic setting tags for S3 broker to match [aws-broker](https://github.com/cloud-gov/aws-broker/blob/main/services/rds/rdsinstance.go#L298-L302)
- Fix unit tests
- Update to use code.cloudfoundry.org/lager/v3 
- Update to use github.com/pivotal-cf/brokerapi/v10

There are lot of changes in the PR related to compatibility for `github.com/pivotal-cf/brokerapi/v10`. Updating to `github.com/pivotal-cf/brokerapi/v10` and `code.cloudfoundry.org/lager/v3 ` was necessary for compatibility with `github.com/onsi/ginkgo/v2`

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations

There are no real security considerations to the main code changes, which simply change the tag names applied to new S3 buckets. But this PR also updates several dependencies to the latest versions, which keeps us up to date with fixes and security patches

### CODE REVIEW

File: broker/broker.go [hunk 1]
```diff
@@ -458,7 +503,14 @@ func (b *S3Broker) bucketFromPlan(servicePlan ServicePlan) *awss3.BucketDetails
 	return bucketDetails
 }
 
-func (b *S3Broker) bucketTags(action, serviceID, planID, organizationID, spaceID string) map[string]string {
+func getBucketTags(
```
Comment (markdboyd) [inline broker/broker.go:506]: The actual changes related to tags are in this function

File: broker/broker_tags_test.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package broker
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestGetBucketTags(t *testing.T) {
```
Comment (markdboyd) [inline broker/broker_tags_test.go:8]: Unit tests for the new tags logic

File: broker/broker.go [hunk 3]
```diff
@@ -306,7 +313,7 @@ func (b *S3Broker) Bind(
 			bucketDetails, err := b.bucket.Describe(bucketName, b.awsPartition)
 			if err != nil {
 				if err == awss3.ErrBucketDoesNotExist {
-					errc <- brokerapi.ErrInstanceDoesNotExist
+					errc <- apiresponses.ErrInstanceDoesNotExist
```
Comment (rcgottlieb) [inline broker/broker.go:316]: is apiresponse supposed to be domain?

**Review by jameshochadel — APPROVED** [review]
Nice work updating the deps and bringing the tags in line with the aws-broker!

File: broker/broker.go (hunk 3 above)
Comment (markdboyd) [reply broker/broker.go:316]: Nope, the errors are imported from a different package. I was getting errors in VSCode when trying to import from `domain`

**Pull request merged**